iceoryx_binding_c/include/iceoryx_binding_c/posh_runtime.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_BINDING_C_POSH_RUNTIME_H_
+#define IOX_BINDING_C_POSH_RUNTIME_H_
+
+#include "internal/c2cpp_bridge.h"
+
+void PoshRuntime_getInstance(const char* const name);
+
+/// @brief retrieves the instance-name of the current posh runtime instance
+/// @param[in] name char pointer to preallocated memory
+/// @param[in] nameLength size of the preallocated memory.
+/// @return The length of the instance-name. If the instance-name is longer then nameLength a
+///         number greater nameLength is returned and the instance-name, truncated
+///         to nameLength, is written into the memory location of name.
+uint64_t PoshRuntime_getInstanceName(char* const name, const uint64_t nameLength);
+
+void PoshRuntime_findService();
+void PoshRuntime_offerService();
+void PoshRuntime_stopOfferService();
+
+void PoshRuntime_getMiddlewareSender();
+void PoshRuntime_getMiddlewareReceiver();
+void PoshRuntime_getMiddlewareInterface();
+void PoshRuntime_getMiddlewareApplication();
+
+void PoshRuntime_getMiddlewareConditionVariable();
+void PoshRuntime_createRunnable();
+uint64_t* PoshRuntime_getServiceRegistryChangeCounter();
+bool PoshRuntime_sendRequestToRouDi();
```

Comment:
Do all those function really need to be accessible by the user? I thought they are called in c'tor of `Publisher` and `Subscriber`.